--- a/Makefile	2017-10-31 09:39:12.416812633 +0100
+++ b/Makefile	2017-10-31 09:38:11.357482904 +0100
@@ -1,11 +1,11 @@
 PROJECT=pifacedigital
-SOURCES=src/pifacedigital.c
-LIBRARY=static
-INCPATHS=../libmcp23s17/src/
-LIBPATHS=../libmcp23s17/
-LDFLAGS=
-CFLAGS=-c -Wall
-CC=gcc
+SOURCES = src/pifacedigital.c
+LIBRARY = shared
+INCPATHS = /usr/src/
+LIBPATHS = /usr/lib/
+LDFLAGS +=
+CFLAGS += -Wall -lmcp23s17
+CC ?= gcc
 
 # ------------ MAGIC BEGINS HERE -------------
 
@@ -31,11 +31,11 @@
     ifeq "$(LIBRARY)" "static"
 	ar rcs $(BINARY) $(OBJECTS)
     else
-	$(CC) $(LIBFLAGS) $(OBJECTS) $(LDFLAGS) -o $@
+	$(CC) $(CFLAGS) $(LIBFLAGS) $(OBJECTS) $(LDFLAGS) -o $@
     endif
 
 .c.o:
-	$(CC) $(INCFLAGS) $(CFLAGS) -fPIC $< -o $@
+	$(CC) $(INCFLAGS) $(CFLAGS) $(LDFLAGS) $< -o $@
 
 distclean: clean
 	rm -f $(BINARY)

